client/state/login/actions.js

```diff
@@ -278,7 +283,40 @@ export const createSocialUser = ( service, token, flowName ) => dispatch => {
 			error,
 		} );
 
-		return Promise.reject( wpcomError );
+		return Promise.reject( error );
+	} );
+};
+
+/**
+ * Attempt to connect the current account with a social service
+ *
+ * @param  {String}    service    The external social service name.
+ * @param  {String}    token      Authentication token provided by the external social service.
+ * @param  {String}    redirectTo Url to redirect the user to upon successful login
+ * @return {Function}             Action thunk to trigger the login process.
+ */
+export const connectSocialUser = ( service, token, redirectTo ) => dispatch => {
+	dispatch( {
+		type: SOCIAL_CONNECT_ACCOUNT_REQUEST,
+		notice: {
+			message: translate( 'Creating your account' )
+		},
+	} );
+
+	return wpcom.undocumented().usersSocialConnect( service, token, redirectTo ).then( wpcomResponse => {
+		dispatch( {
+			type: SOCIAL_CONNECT_ACCOUNT_REQUEST_SUCCESS,
+			redirectTo: wpcomResponse.redirect_to,
+		} );
```

Comment:
I don't think so, none of the success routines in the other login functions return a `Promise`.